cli/common/socket.go

```diff
@@ -39,6 +39,7 @@ func ReadFromConn(conn net.Conn) ([]byte, error) {
 	data := make([]byte, 0)
 
 	for {
+		conn.SetReadDeadline(time.Now().Add(10 * time.Second))
```

Comment:
How does it work? Could we place this line before the cycle?

Also IMHO 10 seconds it's quite huge timeout for console application. 3-4 seconds are enough but it's up to you

Reply:
I'm not sure about 3-4 second because this time depends on duration of Cartridge operation performed by this call. And it can be long sometimes (new Cartridge joins replica sets synchronously).

`SetReadDeadline` says when (since now) read should fail. I think this line can be placed before cycle, you are right.